### PR TITLE
Non-record: VRL + LeakyReLU² + Full SOTA Stack (iteration build)

### PR DESCRIPTION
**Pull request by AnubhavBharadwaaj**
RUN_ID=smoke2 SEED=1337 ITERATIONS=500 \
  TRAIN_BATCH_TOKENS=131072 TRAIN_SEQ_LEN=2048 \
  VAL_LOSS_EVERY=0 TRAIN_LOG_EVERY=25 \
  MAX_WALLCLOCK_SECONDS=1200 \
  DATA_PATH=./data/datasets/fineweb10B_sp1024/ \
  TOKENIZER_PATH=./data/tokenizers/fineweb_1024_bpe.model \
  VOCAB_SIZE=1024 \
  torchrun --standalone --nproc_per_node=1 \
    records/track_non_record_16mb/2026-03-27_VRL_LeakyReLU2_GPTQ/train_gpt.py \
  2>&1 | tee smoke_test2.log
Building SOTA model: 11L, 512d, 3x MLP, XSA last 4 layers, LeakyReLU²(slope=0.5)
Total parameters: 26,829,912

Starting training: batch=131,072 tokens, seq_len=2048, warmdown=3500
Step 1: Late QAT activated (lr_scale=0.0500)
step=25 | train_loss=5.3126 | lr_scale=0.0448 | step_time=1761.5ms | elapsed=43.3s | qat=ON
step=50 | train_loss=5.1735 | lr_scale=0.0402 | step_time=1265.8ms | elapsed=62.5s | qat=ON
step=75 | train_loss=5.0654 | lr_scale=0.0359 | step_time=1101.2ms | elapsed=81.8s | qat=ON
step=100 | train_loss=5.1499 | lr_scale=0.0319 | step_time=1020.6ms | elapsed=101.3s | qat=ON
step=125 | train_loss=4.9812 | lr_scale=0.0281 | step_time=774.9ms | elapsed=120.8s | qat=ON
^[[A^[[A^[[B^[[B^[[B^[[Bstep=150 | train_loss=4.9898 | lr_scale=0.0245 | step_time=776.1ms | elapsed=140.1s | qat=ON
step=175 | train_loss=5.0093 | lr_scale=0.0211 | step_time=780.2ms | elapsed=159.8s | qat=ON
step=200 | train_loss=4.9182 | lr_scale=0.0180 | step_time=779.7ms | elapsed=179.3s | qat=ON
step=225 | train_loss=4.8911 | lr_scale=0.0152 | step_time=779.2ms | elapsed=198.7s | qat=ON
step=250 | train_loss=4.9016 | lr_scale=0.0125 | step_time=780.6ms | elapsed=218.2s | qat=ON
step=275 | train_loss=4.9626 | lr_scale=0.0102 | step_time=778.0ms | elapsed=237.6s | qat=ON
step=300 | train_loss=4.8946 | lr_scale=0.0080 | step_time=778.3ms | elapsed=257.1s | qat=ON
step=325 | train_loss=4.8770 | lr_scale=0.0062 | step_time=781.4ms | elapsed=276.8s | qat=ON
step=350 | train_loss=4.8561 | lr_scale=0.0045 | step_time=785.8ms | elapsed=296.7s | qat=ON
step=375 | train_loss=4.8868 | lr_scale=0.0031 | step_time=788.6ms | elapsed=316.5s | qat=ON
step=400 | train_loss=4.8844 | lr_scale=0.0020 | step_time=790.2ms | elapsed=336.1s | qat=ON
step=425 | train_loss=4.8747 | lr_scale=0.0011 | step_time=788.0ms | elapsed=355.6s | qat=ON
step=450 | train_loss=4.8459 | lr_scale=0.0005 | step_time=782.9ms | elapsed=375.1s | qat=ON
step=475 | train_loss=4.8406 | lr_scale=0.0001 | step_time=781.0ms | elapsed=394.6s | qat=ON
step=500 | train_loss=4.8391 | lr_scale=0.0000 | step_time=782.5ms | elapsed=414.3s | qat=ON

Applying EMA weights...
Applying SWA (10 checkpoints)...
                                                                                                                                             Final (pre-quant): val_loss=4.9621 | val_bpb=2.9389
Quantizing (int6 MLP/attn, int8 embed, GPTQ-lite)...

Final (pre-quant): val_loss=4.9621 | val_bpb=2.9389
Quantizing (int6 MLP/attn, int8 embed, GPTQ-lite)...
Post-quant roundtrip: val_loss=4.9654 | val_bpb=2.9408                                                                                       Quantization gap: +0.0020 BPB                                              

============================================================
ARTIFACT SUMMARY
============================================================
Code size:              50,152 bytes
Model size:          5,549,512 bytes (zlib-9)
Total artifact:      5,599,664 bytes (5.60 MB)
16MB limit:         16,000,000 bytes
Headroom:           10,400,336 bytes
============================================================
final_val_bpb:    2.9408
final_val_loss:   4.9654
============================================================
Artifact fits within 16MB limit.
Saved: model_smoke2.bin

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Non-record: VRL + LeakyReLU² + Full SOTA Stack (iteration build)

**BPB:** 2.9389 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `c848b65f1ec2`, file `records/track_non_record_16mb/2026-03-27_VRL_LeakyReLU2_GPTQ/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.20s, dim=512, layers=11, vocab=1024, code=50154 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.20s, dim=512, layers=11, vocab=1024, code=50154 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
